### ISSUE TITLE
Items Field is Unsupported

### ISSUE THREAD

**Issue by marcodeluca** [annhdev/strapi-tree-menus#3]
After installing, the items field is disabled with the following placeholder:

Unsupported field type: plugin::tree-menus.tree

**Comment by annhdev** [comment]
@marcodeluca , please provide the version of strapi you are using, and the specific error code for me to review.

**Comment by marcodeluca** [comment]
Thanks for the quick response @annhdev 

    Node.js version: v18.19.0
    NPM version: 10.2.3
    Strapi version: 5.1.1
    Database: Mysql 8+
    Operating system: Ubuntu
    Is your project Javascript or Typescript: Typescript

Not sure how else to describe the bug. I cannot enter menu items unless I am misunderstanding something about how to setup or use the plugin. The items field is disabled.

<img width="789" alt="Screen Shot 2024-10-26 at 7 51 04 AM" src="https://github.com/user-attachments/assets/983ecc28-5a32-443a-9202-5dc9e1e2f60f">

**Comment by marcodeluca** [comment]
and this is what I have in my plugins.ts as per the documentation. So yah, not sure what else I am missing?

```
export default ({ env }) => ({
  menus: {
    enabled: true,
    config: {
      fieldSchema: {
        attributes: [
          {
            id: "title",
            label: "Title",
            placeholder: "Enter item title",
            type: "text",
            validationType: "string",
            value: "New items",
            required: true,
            validations: [
              {
                type: "required",
                params: ["this field is required"]
              },
              {
                type: "max",
                params: [100, "Title cannot be more than 100 characters"]
              },
              {
                type: "default",
                params: ["New items"]
              }
            ]
          },
          {
            id: "url",
            label: "Url",
            placeholder: "Enter url",
            type: "text",
            validationType: "string",
            value: "/",
            required: true,
            validations: [
              {
                type: "required",
                params: ["this field is required"]
              },
              {
                type: "max",
                params: [200, "Url cannot be more than 200 characters"]
              },
              {
                type: "default",
                params: ["/"]
              }
            ]
          },
          {
            id: "target",
            label: "Target",
            placeholder: "Enter target",
            type: "select",
            validationType: "mixed",
            value: '_self',
            required: true,
            validations: [
              {
                type: "oneOf",
                params: [['_blank', '_parent', '_self', '_top'], "this field needs to be one of the following: _blank, _parent, _self, _top"]
              },
              {
                type: "default",
                params: ['_self']
              }
            ],
            options: [
              {
                key: "_blank",
                value: "_blank",
                metadatas: {
                  intlLabel: {
                    id: "tree-menus.target.options._blank",
                    defaultMessage: "New window (_blank)"
                  },
                  disabled: false,
                  hidden: false
                }
              },
              {
                key: "_parent",
                value: "_parent",
                metadatas: {
                  intlLabel: {
                    id: "tree-menus.target.options._parent",
                    defaultMessage: "Parent window (_parent)"
                  },
                  disabled: false,
                  hidden: false
                }
              },
              {
                key: "_self",
                value: "_self",
                metadatas: {
                  intlLabel: {
                    id: "tree-menus.target.options._self",
                    defaultMessage: "Same window (_self)"
                  },
                  disabled: false,
                  hidden: false
                }
              },
              {
                key: "_top",
                value: "_top",
                metadatas: {
                  intlLabel: {
                    id: "tree-menus.target.options._top",
                    defaultMessage: "Top window (_top)"
                  },
                  disabled: false,
                  hidden: false
                }
              }
            ]
          },
          {
            id: "isProtected",
            label: "isProtected",
            placeholder: "Choose isProtected",
            type: "bool",
            validationType: "boolean",
            value: false,
            required: true,
            validations: [
              {
                type: "required",
                params: ["Need to choose isProtected"]
              },
              {
                type: "default",
                params: [false]
              }
            ]
          }
        ]
      },
    },
  },
});
```

**Comment by annhdev** [comment]
@marcodeluca , I found a bug in Vite's build process with the traverse library package. It's a common bug in all Vite-based build systems. I've updated the ReadMe.md with detailed instructions, you can see it here:
[ReadMe](https://github.com/annhdev/strapi-tree-menus?tab=readme-ov-file#hot-fix-issue-between-vite-and-traverse)

**Comment by annhdev** [comment]
> and this is what I have in my plugins.ts as per the documentation. So yah, not sure what else I am missing?
> 
> ```
> export default ({ env }) => ({
>   menus: {
>     enabled: true,
>     config: {
>       fieldSchema: {
>         attributes: [
>           {
>             id: "title",
>             label: "Title",
>             placeholder: "Enter item title",
>             type: "text",
>             validationType: "string",
>             value: "New items",
>             required: true,
>             validations: [
>               {
>                 type: "required",
>                 params: ["this field is required"]
>               },
>               {
>                 type: "max",
>                 params: [100, "Title cannot be more than 100 characters"]
>               },
>               {
>                 type: "default",
>                 params: ["New items"]
>               }
>             ]
>           },
>           {
>             id: "url",
>             label: "Url",
>             placeholder: "Enter url",
>             type: "text",
>             validationType: "string",
>             value: "/",
>             required: true,
>             validations: [
>               {
>                 type: "required",
>                 params: ["this field is required"]
>               },
>               {
>                 type: "max",
>                 params: [200, "Url cannot be more than 200 characters"]
>               },
>               {
>                 type: "default",
>                 params: ["/"]
>               }
>             ]
>           },
>           {
>             id: "target",
>             label: "Target",
>             placeholder: "Enter target",
>             type: "select",
>             validationType: "mixed",
>             value: '_self',
>             required: true,
>             validations: [
>               {
>                 type: "oneOf",
>                 params: [['_blank', '_parent', '_self', '_top'], "this field needs to be one of the following: _blank, _parent, _self, _top"]
>               },
>               {
>                 type: "default",
>                 params: ['_self']
>               }
>             ],
>             options: [
>               {
>                 key: "_blank",
>                 value: "_blank",
>                 metadatas: {
>                   intlLabel: {
>                     id: "tree-menus.target.options._blank",
>                     defaultMessage: "New window (_blank)"
>                   },
>                   disabled: false,
>                   hidden: false
>                 }
>               },
>               {
>                 key: "_parent",
>                 value: "_parent",
>                 metadatas: {
>                   intlLabel: {
>                     id: "tree-menus.target.options._parent",
>                     defaultMessage: "Parent window (_parent)"
>                   },
>                   disabled: false,
>                   hidden: false
>                 }
>               },
>               {
>                 key: "_self",
>                 value: "_self",
>                 metadatas: {
>                   intlLabel: {
>                     id: "tree-menus.target.options._self",
>                     defaultMessage: "Same window (_self)"
>                   },
>                   disabled: false,
>                   hidden: false
>                 }
>               },
>               {
>                 key: "_top",
>                 value: "_top",
>                 metadatas: {
>                   intlLabel: {
>                     id: "tree-menus.target.options._top",
>                     defaultMessage: "Top window (_top)"
>                   },
>                   disabled: false,
>                   hidden: false
>                 }
>               }
>             ]
>           },
>           {
>             id: "isProtected",
>             label: "isProtected",
>             placeholder: "Choose isProtected",
>             type: "bool",
>             validationType: "boolean",
>             value: false,
>             required: true,
>             validations: [
>               {
>                 type: "required",
>                 params: ["Need to choose isProtected"]
>               },
>               {
>                 type: "default",
>                 params: [false]
>               }
>             ]
>           }
>         ]
>       },
>     },
>   },
> });
> ```

sorry for the wrong config readme, I updated the readme, the correct config is "tree-menus": {...}
By default you don't need to add fieldSchema configuration, unless you need to add custom fields

**Comment by marcodeluca** [comment]
Thank you that was the issue and it resolves the issue. FYI - minor issue I see with the drag/drop items now, when you click expand/contract on the item (accordion) it flashes closed and immediately open again.

Also curious if it is possible to add a relationship field, in the documentation it does not specify either that it does or does not accept that type of field. But those are separate issues i can create other issues for those.

Thanks again!

**Comment by annhdev** [comment]
> Thank you that was the issue and it resolves the issue. FYI - minor issue I see with the drag/drop items now, when you click expand/contract on the item (accordion) it flashes closed and immediately open again.
> 
> Also curious if it is possible to add a relationship field, in the documentation it does not specify either that it does or does not accept that type of field. But those are separate issues i can create other issues for those.
> 
> Thanks again!

To be able to create a tree custom field in a table, and support i18n for it, i can only save the tree data as json. So it doesn't support relationship fields.
I designed the menu management part to be as simple as possible, not trying to complicate the problem.
Error collapse flicker, fixed in new version v1.0.4, please upgrade
